packages/components/src/index.ts

```diff
@@ -213,6 +213,7 @@ export {
 } from './higher-order/with-focus-return';
 export { default as withNotices } from './higher-order/with-notices';
 export { default as withSpokenMessages } from './higher-order/with-spoken-messages';
+export { ComponentsContext as __experimentalComponentsContext } from './ui/context/context-system-provider';
```

Comment:
Should we be leveraging the private APIs here for the ComponentsContext instead of exporting it as  `__experimental`?

Reply:
Yup! It is already the case in #51264 (the equivalent code in this PR will be removed once #51264 is merged)